File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    lost_set = set(lost)
    reserve_set = set(reserve)

    # 여벌이 있지만 도난당한 경우 제거
    intersect = lost_set & reserve_set
    lost_set -= intersect
    reserve_set -= intersect

    for student in sorted(lost_set):
        if student - 1 in reserve_set:
            reserve_set.remove(student - 1)
        elif student + 1 in reserve_set:
            reserve_set.remove(student + 1)
        else:
            continue
        lost_set.remove(student)

    return n - len(lost_set)

File: 프로그래머스/1/42862. 체육복/체육복.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        HashSet<int> lostSet = new HashSet<int>(lost);
        HashSet<int> reserveSet = new HashSet<int>(reserve);
        
        HashSet<int> intersect = new HashSet<int>(lostSet);
        intersect.IntersectWith(reserveSet);
        
        foreach (int student in intersect) 
        {
            lostSet.Remove(student);
            reserveSet.Remove(student);
        }
        
        List<int> sortedLost = lostSet.OrderBy(x => x).ToList();
        
        foreach (int student in sortedLost.ToList())
        {
            if (reserveSet.Contains(student - 1)) 
            {
                reserveSet.Remove(student - 1);
                lostSet.Remove(student);
            }
            else if (reserveSet.Contains(student + 1)) 
            {
                reserveSet.Remove(student + 1);
                lostSet.Remove(student);
            }
        }
        
        int answer = n - lostSet.Count;
        return answer;
    }
}